library/src/main/java/com/novoda/downloadmanager/RoomV2DatabaseFiles.java

```diff
@@ -0,0 +1,24 @@
+package com.novoda.downloadmanager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class RoomV2DatabaseFiles implements V2DatabaseFiles {
```

Comment:
I'm not completely sold on this class TBH. 

1. I think the naming is confusing perhaps it could be `RoomFiles` but then it doesn't contain files, just names.
2. Part of me thinks this could just exist on the `RoomAppDatabase` object where you call `roomFileNames()` and it does the `fileNames()` method there instead. 
3. The interface feels unnecessary but I guess was added for testing? But it seems like you could `mock` the `RoomFileDao` or the whole `Database` instead.